clippy_lints/src/methods/mod.rs

```diff
@@ -2530,6 +2555,20 @@ fn lint_filter_next<'a, 'tcx>(
     }
 }
 
+/// lint use of `skip_while().next()` for `Iterators`
+fn lint_skip_while_next<'a, 'tcx>(
+    cx: &LateContext<'a, 'tcx>,
+    expr: &'tcx hir::Expr<'_>,
+    _skip_while_args: &'tcx [hir::Expr<'_>],
+) {
+    // lint if caller of `.skip_while().next()` is an Iterator
+    if match_trait_method(cx, expr, &paths::ITERATOR) {
+        let msg = "called `skip_while(p).next()` on an `Iterator`. \
+                   This is more succinctly expressed by calling `.find(!p)` instead.";
+        span_lint(cx, SKIP_WHILE_NEXT, expr.span, msg);
```

Comment:
Can you split up the message and use `span_help_and_lint`?

Reply:
Yeah, it's reasonable, addressed. And I've realized that some similar lints also use `span_lint`. I'll follow up later.

Reply:
Do you want to do this in this PR or open a new one?